hibernate-core/src/main/java/org/hibernate/spi/NavigablePath.java

```diff
@@ -195,14 +199,14 @@ public boolean isParent(NavigablePath navigablePath) {
 	/**
 	 * Determine whether the given path is a suffix of this path
 	 */
-	public boolean isSuffix(DotIdentifierSequence dotIdentifierSequence) {
+	public boolean isSuffix(@Nullable DotIdentifierSequence dotIdentifierSequence) {
 		if ( dotIdentifierSequence == null ) {
 			return true;
 		}
 		if ( !localNamesMatch( dotIdentifierSequence ) ) {
 			return false;
 		}
-		return getParent() != null && getParent().isSuffix( dotIdentifierSequence.getParent() );
+		return getParent() != null && castNonNull( getParent() ).isSuffix( dotIdentifierSequence.getParent() );
```

Comment:
```suggestion
		DotIdentifierSequence parent = getParent();
		return parent  != null && parent.isSuffix( dotIdentifierSequence.getParent() );
```

Reply:
The reference needs to be a NavigablePath I think?

Reply:
yeah, you're right